src/_factories.ts

```diff
@@ -187,29 +187,32 @@ export function makeWidgetFactory(definition: WidgetDefinition, resolveMid: Reso
 		}
 	}
 
-	let { options } = definition;
-	if (options && ('id' in options || 'listeners' in options || 'stateFrom' in options)) {
+	const { options: rawOptions } = definition;
+	if (rawOptions && ('id' in rawOptions || 'listeners' in rawOptions || 'stateFrom' in rawOptions)) {
 		throw new TypeError('id, listeners and stateFrom options should be in the widget definition itself, not its options value');
 	}
-	options = assign({ id: definition.id }, options);
 
-	return () => {
+	const options: {
+		id: string,
+		listeners?: EventedListenersMap,
+		stateFrom?: StoreLike
+	} = assign({ id: definition.id }, rawOptions);
+
+	return ({ stateFrom: defaultStore }: { stateFrom?: StoreLike }) => {
 		return Promise.all<any>([
```

Comment:
Nah, the generic is the value type for the resulting array. https://github.com/dojo/core/blob/a4e2f7e720250ae1b14f36407eb99ead1452d59b/src/Promise.ts#L48